webpack/assets/javascripts/react_app/components/ToastsListRails/index.js

```diff
@@ -0,0 +1,23 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { notify } from '../../../foreman_toast_notifications';
+import ToastsList from '../ToastsList';
+
+const ToastsListRails = ({ rails_messages: railsMessages }) => {
```

Comment:
can we use this component in the `PageLayout` as well? https://github.com/theforeman/foreman/blob/7c240321be43e600afd790f6c33693effa73f7f0/webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js#L32

Reply:
Why should we use rails notifications in the react PageLayout?

Reply:
Rails notifications should be rendered on the client but are created on the server through `flash`
```js
// PageLayout.js
 <div
     id="toast-notifications-container"
     data-notifications={toastNotifications}
  >
  <ToastsList />
```
Looks like it uses the same code you just deleted (uses the data attirbute `notification`). on each `contentLoad` event it push notification to `toast-notifications-container` and then render all on `ToastsList` component. this change need to be tested on client rendering with full react pages via AJAX.

Reply:
With pageLayout and this PR and without this PR the behavior is the same: rails sends a flash message -> it doesn't show up-> user reloads the page -> toast notification from the flash message shows

Reply:
right, but the code in pageLayout looks like a dead code (` data-notifications` has no use)

Reply:
You're right, removed it